### PR TITLE
MA-14: Create locale menu

### PR DESCRIPTION
**Pull request by vitalii-petrun**
https://user-images.githubusercontent.com/87064627/196028101-4a358e38-5734-4fd8-b52d-bacedf5899ac.mp4

Also, need to be done:
- add highlight color of nav bar items

### CODE REVIEW

File: lib/data/datasources/local/shared_prefs.dart [hunk 1]
```diff
@@ -23,6 +23,7 @@ class SharedPrefs {
 
   Future<bool> setLocale(AppLocale locale) async =>
       _preferences.setString(_localeKey, locale.name);
+
```
Comment (vovaklh) [inline lib/data/datasources/local/shared_prefs.dart:26]: Delete this line

File: lib/l10n/app_ru.arb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    "@_LOGIN_PAGE": {},
+    "authorization": "Авторизация",
+    "yourLogin": "Ваш логин",
+    "yourLoginWithTip": "Ваш логин (имя аккаунта)",
+    "yourPassword": "Ваш пароль",
+    "codeFromImage": "Код с изображения*",
+    "dontHaveAnAccount": "Ещё нет аккаунта?",
+    "signUp": "Зарегистрироваться",
+    "forgotPassword": "Забыли пароль?",
+    "restore": "Возобновить",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:11]: Maybe Восстановить

File: lib/l10n/app_ru.arb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    "login": "Логин",
+    "incorrectLogin": "Неправильный логин",
+    "incorrectPassword": "Неправильный пароль",
+    "incorrectCode": "Не корректный код",
+    "fieldMustNotBeEmpty": "Это поле не должно быть пустым",
+    "passwordMustBeLonger": "Пароль должен быть длиннее {minLength}и символов",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:17]: Why "и" after minLength

File: lib/l10n/app_ru.arb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    "@passwordMustBeLonger": {
+        "placeholders": {
+            "minLength": {
+                "type": "int"
+            }
+        }
+    },
+    "@_FORGOT_PASSWORD_PAGE": {},
+    "accessRecovery": "Получить восстановление ???",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:26]: Check this

File: lib/l10n/app_ru.arb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    "logInSystem": "Авторизация в системе",
+    "registrationEmail": "Электронная почта для регистрации",
+    "restoreAccess": "Возобновить доступ",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:29]: Check this

File: lib/l10n/app_ru.arb [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    "@_HOME_PAGE": {},
+    "blog": "Блог",
+    "moreFactsAbout": "Больше фактов о Мультимета в обновлённой",
+    "multiMetaUniverse": "MultiMeta Universe",
+    "metaverseThatUnites": "Метавселенная, которая объединяет",
+    "profitToPartners": "Прибыль партнерам от 0.8% в день",
+    "leadershipBonuses": "Лидерские бонусы в 15 уровней",
+    "fromFiftyDollars": "От $50 дo $1,000,000 за партнерский ранг для лидеров",
+    "account": "Аккаунт",
+    "roadmap": "Дорожной карте",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:48]: Check this

File: lib/l10n/app_ru.arb [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    "products": "Продукты",
+    "partners": "Партнеры",
+    "@_FOOTER": {},
+    "weCareAboutYou": "Мы заботимся о Вас | MultiMeta Universe  | Окленд",
+    "telegram": "Telegram",
+    "email": "Электронная почта",
+    "submitTicket": "Отправить билет",
+    "questions": "Вопросы",
+    "usagePolicy": "Политика использования",
+    "whitePaper": "White paper",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:61]: Check this

File: lib/l10n/app_ru.arb [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    "officeProgram": "Office program",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:62]: Check this

File: lib/l10n/app_ru.arb [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    "multimeta": "MultiMeta",
+    "resources": "Ресурсы",
+    "mediaAndSupport": "Медиа и поддержка",
+    "@_LOCALE_DROPDOWN": {},
+    "english": "English",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:67]: Check this

File: lib/l10n/app_uk.arb [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    "whitePaper": "Білий папір",
+    "officeProgram": "Офісна програма",
+    "multimeta": "MultiMeta",
+    "resources": "Ресурси",
+    "mediaAndSupport": "Медіа та підтримка",
+    "@_LOCALE_DROPDOWN": {},
+    "english": "English",
+    "russian": "Российский",
+    "ukrainian": "Українська"
+}
```
Comment (vovaklh) [inline lib/l10n/app_uk.arb:70]: Check like file below

File: lib/presentation/themes/src/theme_colors.dart [hunk 11]
```diff
@@ -11,6 +11,9 @@ class ThemeColors extends ThemeExtension<ThemeColors> {
   final Color headerBackground;
   final Color headerShadow;
   final Color footerBackground;
+  final Color localeDropdownOpenedBackground;
+  final Color localeDropdownOpacity;
+  final Color localeDropdownBackground;
```
Comment (vovaklh) [inline lib/presentation/themes/src/theme_colors.dart:16]: I think you should delete this

File: lib/presentation/widgets/header/components/locale_dropdown.dart [hunk 12]
```diff
@@ -7,16 +7,185 @@ class _LocaleDropdown extends StatefulWidget {
...
+      duration: const Duration(milliseconds: 500),
+    );
+    animation = CurveTween(curve: Curves.linear).animate(animationController);
+  }
+
+  void showOverlay() {
+    final overlayState = Overlay.of(context);
+    final renderBox = context.findRenderObject() as RenderBox;
+    final offset = renderBox.localToGlobal(Offset.zero);
+    final compactMenuDxPosition = renderBox.size.width / 2 - 70;
```
Comment (vovaklh) [inline lib/presentation/widgets/header/components/locale_dropdown.dart:30]: Delete this

File: lib/presentation/widgets/header/components/locale_dropdown.dart [hunk 13]
```diff
@@ -7,16 +7,185 @@ class _LocaleDropdown extends StatefulWidget {
...
+
+    _overlayEntry = OverlayEntry(
+      builder: (context) => Positioned(
+        top: offset.dy,
+        left: context.screenWidth > 780 ? offset.dx : compactMenuDxPosition,
```
Comment (vovaklh) [inline lib/presentation/widgets/header/components/locale_dropdown.dart:35]: Refactor

File: lib/presentation/widgets/header/components/locale_dropdown.dart [hunk 14]
```diff
@@ -7,16 +7,185 @@ class _LocaleDropdown extends StatefulWidget {
...
+
+    switch (locale) {
+      case AppLocale.en:
+        return AppAssets.engFlag;
+      case AppLocale.ru:
+        return AppAssets.rusFlag;
+      case AppLocale.uk:
+        return AppAssets.ukrFlag;
+    }
+  }
```
Comment (vovaklh) [inline lib/presentation/widgets/header/components/locale_dropdown.dart:74]: Create method in enum and try to use https://pub.dev/packages/locale_emoji_flutter insted images

File: lib/presentation/widgets/header/components/locale_dropdown.dart [hunk 15]
```diff
@@ -7,16 +7,185 @@ class _LocaleDropdown extends StatefulWidget {
...
   Widget build(BuildContext context) {
-    return Row(
-      mainAxisAlignment: MainAxisAlignment.center,
-      children: [
-        Image.asset(AppAssets.engFlag, height: 20),
-        const SizedBox(width: 10),
-        const Icon(Icons.arrow_drop_down),
-      ],
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 8.0),
```
Comment (vovaklh) [inline lib/presentation/widgets/header/components/locale_dropdown.dart:178]: Replace by 8

File: lib/presentation/widgets/header/components/locale_dropdown.dart [hunk 16]
```diff
@@ -7,16 +7,185 @@ class _LocaleDropdown extends StatefulWidget {
...
+      child: InkWell(
```
Comment (vovaklh) [inline lib/presentation/widgets/header/components/locale_dropdown.dart:179]: Move Inkwell to the top of widget tree

File: lib/presentation/widgets/header/components/locale_dropdown.dart [hunk 17]
```diff
@@ -7,16 +7,185 @@ class _LocaleDropdown extends StatefulWidget {
...
+
+  @override
+  void initState() {
+    super.initState();
+    animationController = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 500),
+    );
+    animation = CurveTween(curve: Curves.linear).animate(animationController);
+  }
```
Comment (vovaklh) [inline lib/presentation/widgets/header/components/locale_dropdown.dart:24]: Move initState and dispose before build method

**Review by vovaklh — CHANGES_REQUESTED** [review]

File: lib/l10n/app_locale.dart [hunk 18]
```diff
@@ -12,6 +16,24 @@ enum AppLocale {
         return const Locale('en');
       case AppLocale.ru:
         return const Locale('ru');
+      case AppLocale.uk:
+        return const Locale('uk');
     }
   }
+
+  static String? get currentLocaleFlag {
+    final locale = locator<SharedPrefs>().getLocale() ?? AppLocale.en;
```
Comment (vovaklh) [inline lib/l10n/app_locale.dart:25]: Move to the page

File: lib/l10n/app_locale.dart [hunk 19]
```diff
@@ -12,6 +16,24 @@ enum AppLocale {
...
+      case AppLocale.ru:
+        return le.getFlagEmoji(languageCode: 'ru');
+      case AppLocale.uk:
+        return le.getFlagEmoji(languageCode: 'uk');
+    }
+  }
+
+  static String get enFlag => le.getFlagEmoji(languageCode: 'en') as String;
+  static String get ruFlag => le.getFlagEmoji(languageCode: 'ru') as String;
+  static String get ukFlag => le.getFlagEmoji(languageCode: 'uk') as String;
```
Comment (vovaklh) [inline lib/l10n/app_locale.dart:38]: Delete this

File: lib/l10n/app_ru.arb [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+    "@_HOME_PAGE": {},
+    "blog": "Блог",
+    "moreFactsAbout": "Больше фактов о Мультимета в обновлённой",
+    "multiMetaUniverse": "MultiMeta Universe",
+    "metaverseThatUnites": "Метавселенная, которая объединяет",
+    "profitToPartners": "Прибыль партнерам от 0.8% в день",
+    "leadershipBonuses": "Лидерские бонусы в 15 уровней",
+    "fromFiftyDollars": "От $50 дo $1,000,000 за партнерский ранг для лидеров",
+    "account": "Аккаунт",
+    "roadmap": "Дорожной карте",
```
Comment (vovaklh) [inline lib/l10n/app_ru.arb:48]: Maybe "карта"

File: lib/l10n/app_uk.arb [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+    "alreadyHaveAccount": "У мене вже є обліковий запис (Увійти)",
+    "@_HOME_PAGE": {},
+    "moreFactsAbout": "Більше фактів про MultiMeta в оновленій",
+    "multiMetaUniverse": "MultiMeta Universe",
+    "metaverseThatUnites": "Метавсесвіт, який об'єднує",
+    "profitToPartners": "Прибуток партнерам від 0,8% на день",
+    "leadershipBonuses": "Бонуси за лідерство на 15 рівнях",
+    "fromFiftyDollars": "Від $50 до $1 000 000 для Leader Partner Park",
+    "account": "Обліковий запис",
+    "roadmap": "Дорожній карті",
```
Comment (vovaklh) [inline lib/l10n/app_uk.arb:47]: Same as above

File: lib/presentation/widgets/header/components/compact_menu.dart [hunk 22]
```diff
@@ -154,7 +154,7 @@ class _MenuTabs extends StatelessWidget {
         const SizedBox(height: 26),
         const _AccountButton(),
         const SizedBox(height: 26),
-        const _LocaleDropdown(),
+        const SizedBox(width: 82, child: _LocaleDropdown()),
```
Comment (vovaklh) [inline lib/presentation/widgets/header/components/compact_menu.dart:157]: Add comma pls

**Review by vovaklh — CHANGES_REQUESTED** [review]

File: lib/l10n/app_ru.arb (hunk 20 above)
Comment (vitalii-petrun) [reply lib/l10n/app_ru.arb:48]: no, check UI

**Pull request merged**